src/pages/wallet-overview/wallet-overview.spec.ts

```diff
@@ -514,5 +456,493 @@ describe("WalletOverviewPage", () => {
+
+    spyOn(walletBalancesService, "getWalletBalance").and.returnValue(Promise.reject(""));
+    spyOn(alertController, "create").and.returnValue(alert);
+    spyOn(alert, "present");
+
+    comp.getWalletBalance("").then(data => {
+      expect(alertController.create).toHaveBeenCalled();
+      expect(alert.present).toHaveBeenCalled();
+      done();
+    });
+  });
+
+  it("should set the currencies correct for the picked exchange and picked currency and when the currency doesn't exist on the exchange it should pick the first one on the exchange", () => {
+    spyOn(comp, "setCalculatedCurrencyValue");
+
+    comp.pickedExchange = "Exchange1";
+    comp.pickedCurrency = "Curr3";
+    comp.availableExchanges = [
+      {"exchangeName": "Exchange1", "availableCurrencies": ["Curr1", "Curr2", "Curr3", "Curr4"]},
+      {"exchangeName": "Exchange2", "availableCurrencies": ["Curr1", "Curr2", "Curr3"]},
+      {"exchangeName": "Exchange3", "availableCurrencies": ["Curr1", "Curr2"]},
+      {"exchangeName": "Exchange4", "availableCurrencies": ["Curr1", "Curr2"]},
+      {"exchangeName": "Exchange5", "availableCurrencies": ["Curr1", "Curr2", "Curr3", "Curr4"]}
+    ];
+
+    comp.setExchange();
+
+    expect(comp.setCalculatedCurrencyValue).toHaveBeenCalled();
+    expect(comp.pickedCurrency).toBe("Curr3");
+    expect(comp.currentExchangeCurrencies).toEqual(["Curr1", "Curr2", "Curr3", "Curr4"]);
+
+    comp.pickedCurrency = "DON'T EXIST";
+
+    comp.setExchange();
+
+    expect(comp.pickedCurrency).toBe("Curr1");
+  });
+
+  it("should refresh the balance and history on wallet change", () => {
+    spyOn(comp, "getWalletBalance");
+    spyOn(comp, "getTransactionHistory");
+
+    comp.currentWallet = getDummyWallet();
+    comp.onWalletChanged();
+
+    expect(comp.getWalletBalance).toHaveBeenCalledWith("S5NEKHPKXS7F75IVKGVS4A56U4FF6VM5U4YF64");
+    expect(comp.getTransactionHistory).toHaveBeenCalledWith("S5NEKHPKXS7F75IVKGVS4A56U4FF6VM5U4YF64");
+  });
+
+  it("should create a new chart with correct legend", () => {
+    comp.currenciesForDoughnutCanvasLabels = ["Label1", "Label2", "Label3"];
+    comp.currenciesForDoughnutCanvas = [10, 20, 30];
+
+    var dummyElement = document.createElement('canvas');
+    document.getElementById = jasmine.createSpy('HTML Element').and.returnValue(dummyElement);
+    comp.doughnutCanvas = new ElementRef("");
+    comp.doughnutCanvas.nativeElement = document.getElementById("");
+    comp.displayChart();
+    expect(comp.doughnutChart).toBeDefined();
+    let legend = comp.doughnutChart.generateLegend();
+    expect(legend[0]).toEqual({backgroundColor: '#064C70', label: 'Label1', data: 10});
+    expect(legend[1]).toEqual({backgroundColor: '#1B79A9', label: 'Label2', data: 20});
+    expect(legend[2]).toEqual({backgroundColor: '#d0ff00', label: 'Label3', data: 30});
+  });
+
+  it("should not try to generate the dougnut chart because needed information is missing for the chart", () => {
+    comp.displayChart();
```

Comment:
I'm assuming this test works right now because by default the needed information is missing. However is there a chance this might change in the future? If so it might be an idea to explicitely set the needed information to some invalid value.